### PR TITLE
[[ Bug 19901 ]] Statically link Linux server engines to libstdc++

### PR DESCRIPTION
**Pull request by livecodemichael**

### CODE REVIEW

**Comment by runrevmark** [comment]
@livecodemichael You'll also need to check that the externals (that use C++) are statically linked to stdc++ too

**Comment by runrevmark** [comment]
@livecodemichael Great - hopefully this will solve the issue! :)

@livecode-vulcan review ok 9e44466

**Comment by livecode-vulcan** [comment]
💙 review by @runrevmark ok 9e444663e4ebbe9ac6ac44c96ff72560bbfe4d7a

**Comment by livecode-vulcan** [comment]
😞 test failure 9e444663e4ebbe9ac6ac44c96ff72560bbfe4d7a

* try-community-armv6-android-api8: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/617)
* try-community-armv6-android-api9: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/633)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/607)
* try-community-universal-ios-iphoneos10.3: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos10.3/builds/550)
* try-community-universal-ios-iphonesimulator10.3: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator10.3/builds/553)
* try-community-universal-mac-macosx10.6: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/611)
* try-community-universal-mac-macosx10.9: [failure](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/631)
* try-community-x86-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/616)
* try-community-x86-linux-debian8: [failure](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/633)
* try-community-x86_64-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/615)
* try-community-x86_64-linux-debian8: [failure](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/626)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/648)
* try-community-x86_64-win32: [pending](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/612)

**Comment by runrevmark** [comment]
@livecodemichael Something odd is going on here - the linux and mac builds are failing because odbc driver isn't compiling as it can't find sql.h

**Comment by livecodeali** [comment]
It's because the dbodbc-server target already has a conditions key which is being clobbered / clobbering the added one - they need merging

**Comment by livecodemichael** [comment]
@livecodeali Good spot! That should be the two blocks merged.

**Comment by livecodeali** [comment]
@livecode-vulcan review ok 53ef184

**Comment by livecode-vulcan** [comment]
💙 review by @livecodeali ok 53ef1848c66ac0981fa83dd51e9f16c3f91dcf28

**Comment by livecode-vulcan** [comment]
😞 test failure 53ef1848c66ac0981fa83dd51e9f16c3f91dcf28

* try-community-armv6-android-api8: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/644)
* try-community-armv6-android-api9: [failure](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/660)
* try-community-js-emscripten-sdk1.35: [pending](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/634)
* try-community-universal-ios-iphoneos10.3: [failure](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos10.3/builds/577)
* try-community-universal-ios-iphonesimulator10.3: [failure](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator10.3/builds/580)
* try-community-universal-mac-macosx10.6: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/638)
* try-community-universal-mac-macosx10.9: [failure](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/658)
* try-community-x86-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/643)
* try-community-x86-linux-debian8: [failure](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/660)
* try-community-x86_64-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/642)
* try-community-x86_64-linux-debian8: [failure](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/653)
* try-community-x86-win32: [failure](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/675)
* try-community-x86_64-win32: [pending](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/639)

File: revdb/revdb.gyp [hunk 1]
```diff
@@ -278,12 +292,22 @@
 							'../thirdparty/libiodbc/libiodbc.gyp:libiodbc',
 						],
 					},
+					'OS == "win"',
```
Comment (runrevmark) [inline revdb/revdb.gyp]: I'm not sure this is still quite right... Gyp conditions need to be laid out as follows:
```
'conditions':
[
  [
    'condition expr',
    {
      consquent dict
    },
    {
      alternate dict
    },
  ],
  [
    '<next condition expr>',
    ...
  ],
  ...
],
```

Comment (livecodemichael) [reply revdb/revdb.gyp]: Arghh! I'll get this right at some point.

**Comment by runrevmark** [comment]
@livecode-vulcan review ok 6e49dd2

**Comment by livecode-vulcan** [comment]
💙 review by @runrevmark ok 6e49dd2a2be426407c8f97bab511a6ddc370044a

**Comment by livecode-vulcan** [comment]
😎 test success 6e49dd2a2be426407c8f97bab511a6ddc370044a

* try-community-armv6-android-api8: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/646)
* try-community-armv6-android-api9: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/662)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/636)
* try-community-universal-ios-iphoneos10.3: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos10.3/builds/579)
* try-community-universal-ios-iphonesimulator10.3: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator10.3/builds/582)
* try-community-universal-mac-macosx10.6: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/640)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/660)
* try-community-x86-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/645)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/662)
* try-community-x86_64-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/644)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/655)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/677)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/641)

**Pull request merged**